src/include/duckdb/execution/operator/scan/physical_table_scan.hpp

```diff
@@ -33,6 +33,10 @@ class PhysicalTableScan : public PhysicalOperator {
 	//! The table filters
 	unique_ptr<TableFilterSet> table_filters;
 
+	string table_name;
+
+	bool duplicate_table;
```

Comment:
Why do you have a ```bool duplicate_table``` on the physical table node?
What you probably want is to have a SHARED_HASH_JOIN logical node, where during the optimization phase, when detecting that a previous join would be executed on the same LHS table, you replace it the whole join, and just have a reference to the LHS table name and one children representing the RHS, or something on this line, no?